File: solutions/2551-apply-operations-to-an-array/solution.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n = len(nums)
        output = []

        for i in range(0,n):
            if i < n - 1 and nums[i] == nums[i + 1]:
                nums[i] = nums[i] * 2
                nums[i + 1] = 0

            if nums[i] != 0:
                output.append(nums[i])

        while(len(output) < n):
            output.append(0)

        return output

File: solutions/2551-apply-operations-to-an-array/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var applyOperations = function(nums) {
    const output = new Array(nums.length).fill(0);
    let out_index = 0;

    for(let i = 0; i < nums.length; i++){
        if(nums[i] === nums[i+1]){
            nums[i] = nums[i] * 2;
            nums[i + 1] = 0;
        }

        if(nums[i] !== 0){
            output[out_index] = nums[i];
            out_index += 1;
        }
    }

    return output;
};
